File: factorial/factorial.js
// Implementing recursive solution
module.exports = function factorial (number) {
  if(number<=1) return 1;
  return number*(factorial(number - 1);
}

// Implementing iterative solution
module.exports = function (number){
   var result = 1 ;
   for(i=1; i<=number; i++){
     result *= i;
   }
   return result;
}

//Implementing iterative solution 2 (going backwards in loop)

module.exports = function (number){
   var result = 1 ;
   for(i=number; i>1; i++){
     result *= i;
   }
   return result;
}

File: factorial/factorial.rb
class Integer

  # Iterative
  def factorial_simple number
    fat = 1
    for n in 2..number do 
      fat = fat * n
    end
    fat
  end

  # Iterative Reverse Loop
  def factorial_reverse number
    (n-1).downto(1).each {|i| n *= i}
    n
  end

  # Recursive - Ternary Operator
  def factorial_recursive number
    (number == 1)? 1: (number * factorial_recursive(number-1))      
  end

  # The Ruby-iest version, using reduce function
  def factorial_reduce(number)
    (1..number).reduce(:*)
  end

  alias_method :factorial :factorial_reduce
end